symfony/plugins/orangehrmPimPlugin/Api/EmployeeSkillAPI.php

```diff
@@ -0,0 +1,355 @@
+
+        return new EndpointGetAllResult(
+            EmployeeSkillModel::class,
+            $employeeSkills,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_EMP_NUMBER => $empNumber,
+                    CommonParams::PARAMETER_TOTAL => $this->getEmployeeSkillService()->getEmployeeSkillDao()->getSearchEmployeeSkillsCount(
+                        $employeeSkillSearchParams
+                    )
+                ]
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getEmpNumberRule(),
+            new ParamRule(self::FILTER_COMMENTS),
+            new ParamRule(self::FILTER_YEARS_OF_EXP),
+            ...$this->getSortingAndPaginationParamsRules(EmployeeSkillSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function create(): EndpointCreateResult
+    {
+        $employeeSkill = $this->saveEmployeeSkill();
+        return new EndpointCreateResult(
+            EmployeeSkillModel::class, $employeeSkill,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_EMP_NUMBER => $employeeSkill->getEmployee()->getEmpNumber(),
+                    self::PARAMETER_SKILL_ID => $employeeSkill->getSkill()->getId()
+                ]
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(self::PARAMETER_SKILL_ID, new Rule(Rules::REQUIRED)),
+            $this->getEmpNumberRule(),
+            ...$this->getCommonBodyValidationRules(),
+        );
+    }
+
+    /**
+     * @return ParamRule[]
+     */
+    private function getCommonBodyValidationRules(): array
+    {
+        return [
+            new ParamRule(
+                self::PARAMETER_COMMENTS,
+                new Rule(Rules::STRING_TYPE),
+                new Rule(Rules::LENGTH, [null, self::PARAM_RULE_COMMENTS_MAX_LENGTH]),
+            ),
+            new ParamRule(
+                self::PARAMETER_YEARS_OF_EXP,
+                new Rule(Rules::INT_TYPE),
+                new Rule(Rules::LENGTH, [null, self::PARAM_RULE_YEARS_OF_EXP_MAX_LENGTH]),
+            ),
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function update(): EndpointUpdateResult
+    {
+        $employeeSkill = $this->saveEmployeeSkill();
+
+        return new EndpointUpdateResult(
+            EmployeeSkillModel::class, $employeeSkill,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_EMP_NUMBER => $employeeSkill->getEmployee()->getEmpNumber(),
+                    self::PARAMETER_SKILL_ID => $employeeSkill->getSkill()->getId()
```

Comment:
keep `CommonParams::PARAMETER_EMP_NUMBER` in meta, but remove `self::PARAMETER_SKILL_ID` in all Endpoint operations since it should be in Model